### PR TITLE
[markdown][pdf] drag/drop and copy/paste support for PDF files

### PR DESCRIPTION
**Pull request by Olimbia23**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Feature request: #2509

### Description
- added pdf into the accepted file extension.
- link generation for pdf with the corresponding file name.

### Steps for Testing

1. Log in to Artemis
2. Drag & Drop in any markdown editor component

### Screenshots

demo

### Issue
Issue with copy/paste of pdf files. 
The clipboard sees any file copy/pasted as `image.png` (instead of `file.pdf`) as reported [here](https://stackoverflow.com/questions/34508763/why-chrome-convert-the-image-data-to-image-png-mime-type-when-paste) and [here](https://ourcodeworld.com/articles/read/491/how-to-retrieve-images-from-the-clipboard-with-javascript-in-the-browser) and [here](https://stackoverflow.com/questions/58924013/javascript-paste-event-getting-file-name)
After my research I believe there is no way to fix this problem. However, I am very pleased for suggestions.

P.s. this issue occurs in Github Markdown editor as well.

### CODE REVIEW

File: src/main/webapp/app/shared/markdown-editor/markdown-editor.component.ts [hunk 1]
```diff
@@ -442,7 +442,13 @@ export class MarkdownEditorComponent implements AfterViewInit {
...
                 this.fileUploaderService.uploadMarkdownFile(file).then(
                     (res) => {
-                        const textToAdd = `![${file.name}](${res.path})\n`;
+                        let textToAdd;
+                        if (extension === 'pdf') {
+                            textToAdd = `[${file.name}](${res.path})\n`;
+                        } else {
+                            textToAdd = `![${file.name}](${res.path})\n`;
+                        }
+
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/markdown-editor/markdown-editor.component.ts:451]:
```suggestion
                        let textToAdd = `[${file.name}](${res.path})\n`;
                        if (extension !== 'pdf') {
                            textToAdd = '!' + textToAdd ;
                        }

```

**Comment by Olimbia23** [comment]
@joschahenningsen sorry that I sneaked in for this task 🙈 I would like you to review it :)

**Review by joschahenningsen — APPROVED** [review]
Agree with @Strohgelaender s suggestion. Looks good other than that.

> The clipboard sees any file copy/pasted as image.png (instead of file.pdf)

You probably could get the mimetype somewhat hacky from the api but I wouldn't worry about it that much.

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Since this PR is from an external branch it's kind of hard to test.
Locally it seems like the markdown file API always sends the content type text/html which makes downloading/viewing the uploaded file impossible.
The endpoint `GET /files/markdown/{filename:.+}` might need to be updated.

Using `buildFileResponse` instead of `responseEntityForFilePath` could do the trick.

**Comment by Olimbia23** [comment]
@joschahenningsen thank you for pointing this out and for helping to fix the issue. Just pushed the fix.

**Review by joschahenningsen — APPROVED** [review]
Seems fine to me.

**Review by Strohgelaender — APPROVED** [review]
Tested locally, works as expected.

**Comment by joschahenningsen** [comment]
Hi @Olimbia23, any reason you closed this?/ Do you want me to reopen it from a internal branch?

**Comment by Olimbia23** [comment]
> Hi @Olimbia23, any reason you closed this?/ Do you want me to reopen it from a internal branch?

Hi @joschahenningsen, I closed it so I can reopen from an internal branch and see in bamboo why client-test fails.

**Pull request closed without merging**